### PR TITLE
Fix mem limit

### PR DESCRIPTION
**Pull request by seansan**
Described here: https://magento.com/tech-resources/bug-tracking/issue/index/id/1540/

### CODE REVIEW

**Comment by sreichel** [comment]
Original Post:

__Steps to reproduce:__

in `app/code/core/Mage/Catalog/Model/Product/Image.php`, on line 186,

    if (!isSet($memoryLimit[0])) {

incorrectly refers to $memoryLimit as an array with index [0]. The variable is not an array, so the if statement always evaluates to true and the memoryLimit is ALWAYS interpreted to be 128M regardless of the setting in php.ini

    if (!isSet($memoryLimit[0])) {
        $memoryLimit = "128M";
    }

Corrected code below:

    if (!isSet($memoryLimit)) {
        $memoryLimit = "128M";
    }

__Expected Result:__

Memory limit should be pulled from php.ini from ini

__Actual Result:__

always evaluates to true and uses 128M hardcoded setting from code.

---

@seansan This is not a bug ... it's absolutly corret! `$memoryLimit[0]` gives you the first char of $memoryLimit __string__. If `ini_get('memory_limit')` returns an empty string limit is set to 128M.

With this PR `!isSet($memoryLimit)` would be `false` because it is set - it's just an empty string - and return value would be `string '' (length=0)`

See: http://php.net/manual/en/function.isset.php (Example #1)

So please check code before PR :)

File: app/code/core/Mage/Catalog/Model/Product/Image.php [hunk 1]
```diff
@@ -183,7 +183,7 @@ protected function _getMemoryLimit()
     {
         $memoryLimit = trim(strtoupper(ini_get('memory_limit')));
 
-        if (!isSet($memoryLimit[0])){
+        if (!isSet($memoryLimit)){
```
Comment (LeeSaferite) [inline app/code/core/Mage/Catalog/Model/Product/Image.php]: I don't think this is doing what you think it's doing. 
`$memoryLimit = trim(strtoupper(ini_get('memory_limit')));` will always set the variable. As far as I can tell, `!empty($memoryLimit)` is the proper check in this case.

**Comment by seansan** [comment]
@sreichel right you are
So inderdaad seems to be correct (or at least checking for that first char)

File: app/code/core/Mage/Catalog/Model/Product/Image.php (hunk 1 above)
Comment (seansan) [inline app/code/core/Mage/Catalog/Model/Product/Image.php]: Maybe keep it the way it is?

Comment (sreichel) [inline app/code/core/Mage/Catalog/Model/Product/Image.php]: For me ... I see no need to change this.

Comment (colinmollenhour) [inline app/code/core/Mage/Catalog/Model/Product/Image.php]: The value of `isset()` on an empty string is `true`. This should be `if (empty($memoryLimit)) {`.

```
php > $a = '';
php > var_dump(isset($a));
php shell code:1:
bool(true)
```

**Pull request closed without merging**